### DISCUSSION TITLE
Accessibility of project

### DISCUSSION THREAD

**Discussion by dvnrsn** [ShafSpecs/remix-pwa#152] [General]
This is just some minor feedback. Not complaining just remarking.

I really want to use a project like this but I find it hard to grok in terms of getting started and getting a working PWA. The docs, while not sparse, are a bit meandering and don't really paint a clear path to getting a simple PWA working. I also lack confidence after searching comments in discord.

![CleanShot 2024-09-17 at 23 11 09](https://github.com/user-attachments/assets/16413c56-3c7b-4354-9a0d-86003045bda4)

The project seems to lack an active champion(s) to help folks get going. Here are a few questions I struggle with off the bat.

1. If a service worker is used as a kind of middleware between client and server, what does that mean for an SSR remix app where all the data is acquired by loaders that are hitting a DB? The [defaultFetchHandler](https://remix-pwa.run/docs/main/default-fetch-handler#using-the-default-fetch-handler) seems to indicate an alternate server host.
2. How can I quickly spin up a PWA? Is there a simple demo project?
3. I keep getting an application error in the docs and navigation breaking (Firefox)
4. Can the documentation on deployment have some elaboration?

For now I'm going to shift over to vite-pwa and try to get my bearings. Perhaps I will come back to this project once I have a better understanding of PWAs with react.

**Comment by ShafSpecs** [comment]
Regarding your initial remark, fairs. Remix PWA has a channel in #showcase (on Discord) where I get notified whenever comments been made. I don't actively search the help forum as issues gets buried fast and I never get mentioned in them.

To answer number 1, in the case of an SSR Remix app, the service worker sits between loaders and your default components. If I have the following:
```ts
// db.ts

// initialise prisma, seed db and provide interfaces

export const prisma = prismaclient.$connect()
```
In your loader:
```ts
// app/routes/app.tsx
export const loader = async (...) => {
  const user = await primsa.users.findUnique(...)

  return json({ user })
}
```
The service worker is intercepting the user object. Whatever happens on the server is not made aware to the service worker, only whatever is returned.

Regarding your second query, no. After Remix moved to Vite, I took down the scaffold CLI because it was just a plugin to add. Now that you mention it though, I would bring it back. Although, you wouldn't be able to use it in an existing project.

Regarding your third issue, what's the error about?

And for your last request, yes. I would work on that soon enough 👍
